lib/mini_magick/shell.rb

```diff
@@ -26,14 +26,14 @@ def run(command, options = {})
     end
 
     def execute(command)
-      stdout, stderr, status =
-        MiniMagick.logger.debug(command.join(" ")) do
-          Timeout.timeout(MiniMagick.timeout) do
-            send("execute_#{MiniMagick.shell_api.gsub("-", "_")}", *command)
-          end
+      stdout, stderr, status = nil, nil, nil
+      MiniMagick.logger.debug(command.join(" ")) do
+        Timeout.timeout(MiniMagick.timeout) do
+          stdout, stderr, status = send("execute_#{MiniMagick.shell_api.gsub("-", "_")}", *command)
         end
+      end
 
-      [stdout, stderr, status.exitstatus]
+      [stdout, stderr, status.try(:exitstatus)]
```

Comment:
`try` method is Active Support have, then pure Ruby doesn't have it.
I seems that MiniMagick is independent of Active Support.
I thought about how it would be no good this way.